File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
#include <unordered_set>

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> s;
        for(auto iter=nums.begin(); iter!=nums.end(); iter++){
            if (s.count(*iter)) return true;
            else s.emplace(*iter);
        }
        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count= Counter(nums)
        freq=[[] for x in range(len(nums)+1)]
        for n,c in count.items():
           freq[c].append(n)
        rlt=[]
        for i in range(len(freq)-1,0,-1):
            for n in freq[i]:
                rlt.append(n)
                if len(rlt)==k: return rlt
        return rlt
